module/applications/actor/base-sheet.mjs

```diff
@@ -1115,6 +1135,39 @@ export default class ActorSheet5e extends ActorSheet {
 
   /* -------------------------------------------- */
 
+  /**
+   * Adjust the preparation mode of a dropped spell depending on the drop location on the sheet.
+   * @param {object} itemData    The item data requested for creation. **Will be mutated.**
+   */
+  _onDropSpell(itemData) {
+    if ( itemData.system.level == 0 || !["npc", "character"].includes(this.document.type) ) return;
+
+    // Determine the section it is dropped on, if any.
+    const mode = this._event.target.closest("div[data-type='spell']")?.dataset;
+
+    // If the spell is dropped on a section, set the mode to be that type.
+    if ( mode && mode.level > 0 ) {
+      itemData.system.preparation.mode = mode["preparation.mode"];
+    }
```

Comment:
One thought I have here is that if a NPC has no spellcasting levels, even cantrips dropped should be changed to innate. Generally sheets that include an `"Innate Spellcasting"` feature will list cantrips in that feature within the `at will` list, so it would make sense that they would be listed with the rest of the innate spells.